src/components/frontend-engine/frontend-engine.tsx

```diff
@@ -76,38 +75,46 @@ const FrontendEngineInner = (props: IFrontendEngineProps) => {
 		return () => subscription.unsubscribe();
 	}, [watch]);
 
-	// ================================================
+	// =============================================================================
 	// HELPER FUNCTIONS
-	// ================================================
-	const handleOnSubmit = (data: any) => {
-		console.log(data);
-		onSubmit && onSubmit();
-	};
-
-	const formatFormId = () => {
-		let formId = "frontend-engine";
+	// =============================================================================
+	useImperativeHandle<Partial<IFrontendEngineRef>, Partial<IFrontendEngineRef>>(ref, () => ({
+		getFormState: () => {
+			return formState;
+		},
+		submit: () => {
+			reactFormHookSubmit(handleSubmit)();
+		},
+	}));
 
-		if (id) formId = `frontend-engine-${id}`;
-		else if (className) formId = `frontend-engine-${className}`;
-
-		return formId;
+	const handleSubmit = (data: TFrontendEngineValues) => {
+		console.log(data); // TODO: remove
+		onSubmit?.(data);
 	};
 
+	// =============================================================================
+	// RENDER FUNCTIONS
+	// =============================================================================
+	const formId = id ? `frontend-engine-${id}` : "frontend-engine";
+
 	return (
-		<StyledForm
-			id={formatFormId()}
-			data-testid={formatFormId()}
-			className={className}
-			noValidate
-			onSubmit={reactFormHookSubmit(handleOnSubmit)}
-		>
-			{fields}
-		</StyledForm>
+		<FormProvider {...formMethods}>
```

Comment:
note: without refs, `FormProvider` is needed for child components to interact with react-hook-form through `useFormProvider`